core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -209,14 +249,22 @@ public void restoreFromBackupLocal() throws Exception {
   }
 
   private void startStopTest(AlluxioMasterProcess master) throws Exception {
+    startStopTest(master, true, true);
+  }
+
+  private void startStopTest(AlluxioMasterProcess master,
+      boolean expectWebServiceStarted, boolean expectMetricsSinkStarted) throws Exception {
     waitForServing(ServiceType.MASTER_RPC);
     waitForServing(ServiceType.MASTER_WEB);
```

Comment:
It's confusing how this waitForServing(WEB) then later `expectWebServiceStarted` is false. There are also "waitForServing" and "waitForReady" and "waitForReadyWeb". It's getting really hard to read. Can you merge these methods into a more readable way?

Reply:
I add a comment in each line.

Reply:
Please rearrange the logic here like moving some more checks into `waitForServing`